File: daemon/brands.py
"""Brand registry + CAC ladder logic (Python mirror of the dashboard's brands.js).
Keep this in sync with dashboard/lib/brands.js. The "id" must also appear in
daemon/config.py BRAND_DIRS, pointing at that brand's folder."""

BRANDS = [
    {
        "id": "example-brand",
        "name": "Example Brand",
        "ad_account_id": "act_REPLACE_ME",   # your Meta ad account (prefix with act_)
        "currency": "USD",
        "target_cac": 15,
        "ladder": {"scale_below": 10, "hold_max": 20, "pause_above": 25},
        "learning_ladder": {"scale_below": 12, "hold_max": 25, "pause_above": 35},
        "active": True,
    },
    # Add more brands here, mirroring dashboard/lib/brands.js.
]


def ladder_verdict(cac, age_days, brand):
    """Return (tier, emoji, label) for a CAC given ad-set age and brand thresholds."""
    if cac is None:
        return ("NODATA", "⚪", "No data")
    learning = age_days is not None and age_days < 15 and brand.get("learning_ladder")
    l = brand["learning_ladder"] if learning else brand["ladder"]
    if cac < l["scale_below"]:
        return ("SCALE", "🟢", "SCALE")
    if cac <= l["hold_max"]:
        return ("HOLD", "🟡", "HOLD")
    if cac > l["pause_above"]:
        return ("PAUSE", "🔴", "PAUSE")
    return ("WATCH", "🟠", "WATCH")

File: dashboard/lib/brands.js
// Brand registry — one entry per Meta ad account you manage.
// Duplicate the example block for each brand. Values come from each brand's Playbook.md.
// IMPORTANT: keep this in sync with daemon/brands.py (and the id must match a folder
// listed in daemon/config.py BRAND_DIRS).

export const BRANDS = [
  {
    id: "example-brand",            // url-safe id; also the key in daemon/config.py BRAND_DIRS
    name: "Example Brand",          // display name shown in the dashboard + briefings
    adAccountId: "act_REPLACE_ME",  // your Meta ad account (prefix with act_)
    currency: "USD",
    targetCac: 15,                  // your target cost per acquisition ($)
    // Post-learning ladder (ad-set age >= 15 days)
    ladder: { scaleBelow: 10, holdMax: 20, pauseAbove: 25 },
    // Learning-phase ladder (days 1-14) — softer thresholds
    learningLadder: { scaleBelow: 12, holdMax: 25, pauseAbove: 35 },
    active: true,                   // set false to keep as a stub (not fetched)
  },
  // Add more brands by copying the block above with new id / name / adAccountId / ladder.
];

// Given a CAC, ad-set age in days, and a brand, return the ladder verdict.
export function ladderVerdict(cac, ageDays, brand) {
  if (cac == null || !isFinite(cac)) return { tier: "NODATA", label: "No data", color: "#6b7280" };
  const l = ageDays != null && ageDays < 15 && brand.learningLadder ? brand.learningLadder : brand.ladder;
  if (!l) return { tier: "NODATA", label: "No ladder", color: "#6b7280" };
  if (cac < l.scaleBelow) return { tier: "SCALE", label: "SCALE", color: "#16a34a" };
  if (cac <= l.holdMax) return { tier: "HOLD", label: "HOLD", color: "#ca8a04" };
  if (cac > l.pauseAbove) return { tier: "PAUSE", label: "PAUSE", color: "#dc2626" };
  return { tier: "WATCH", label: "WATCH", color: "#ea580c" }; // between holdMax and pauseAbove
}
